internal/services/api/get_config_files.go

```diff
@@ -24,10 +35,13 @@ func GetConfigFilesHandlerFunc(base util.AbsolutePath, filesService files.FilesS
 			// Response already returned by ProjectDirFromRequest
 			return
 		}
-		configPath := config.GetConfigPath(projectDir, name)
-		cfg, err := config.FromFile(configPath)
+		configPath := configGetConfigPath(projectDir, name)
+		cfg, err := configFromFile(configPath)
 		if err != nil {
-			if errors.Is(err, fs.ErrNotExist) {
+			if aerr, ok := err.(*types.AgentError); ok && aerr.Code == util.UnknownTOMLKeyCode {
```

Comment:
Previous comment example here, the error code passed back to the client would be something like `util.UnknownTOMLKeyCode` within the config file.